Source/UserManagement/Web/Controllers/DataCollectorsController.cs

```diff
@@ -58,25 +55,50 @@ public IActionResult GetById(Guid id)
             {
                 return Ok(result.Items);
             }
-            //TODO: For this, and the rest of the endpoints: Should probably return something else than 404?
             return new NotFoundResult();
         }
 
         [HttpPost]
-        public IActionResult Register([FromBody] RegisterDataCollector command)
+        public IActionResult Register([FromBody] Read.DataCollectors.DataCollector dataCollector)
         {
-            command.DataCollectorId = Guid.NewGuid();
-            command.IsNewRegistration = true;
-            command.RegisteredAt = DateTimeOffset.UtcNow;
-
+            var command = new RegisterDataCollector
+            {
+                DataCollectorId = Guid.NewGuid(),
+                IsNewRegistration = true,
```

Comment:
I agree, I don't see this as a final solution at all. I was implementing this with the mindset that we could have update functionality by friday, and considering the time constraint I did not want to introduce new commands since that would require a whole new set of validations and other shenanigans...